egs/wsj/s5/steps/combine_trans_dirs.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+# Copyright 2016  Xiaohui Zhang  Apache 2.0.
+# Copyright 2019  SmartAction (kkm)
+# Copyright 2019  manhong wang (marvin)
+
+# This script only combines transform file in the aligments dirs, egs: trans.1,  and
+# validates matching of the utterances and alignments after combining. you would need this fmllr trans
+# files after you combine ali or lat dirs(combine_ali_dirs.sh or combine_lat_dis.sh).
+
+# Begin configuration section.
+cmd=run.pl
+tolerance=10
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging.
+
+[[ -f path.sh ]] && . ./path.sh
+. parse_options.sh || exit 1
+
+export LC_ALL=C
+
+if [[ $# -lt 3 ]]; then
+  cat >&2 <<EOF
+Usage: $0 [options] <data> <dest-dir> <src-dir1> <src-dir2> ...
+ e.g.: $0 data/train exp/tri3_trans_combined exp/tri3_trans_1 exp_tri3_trans_2
+Options:
+ --tolerance <int,%>    # maximum percentage of missing trans
+                        # w.r.t. total utterances in <data> before error is
+                        # reported [10]
+
+Note:we do not checks that certain important files are present and compatible in all
+source directories (phones.txt, tree) here.Because you would run combine_trans_dirs.sh 
+or combine_lat_dis.sh first.
+
+EOF
+  exit 1;
+fi
+
+
+data=$1
+dest=$2
+shift 2
+first_src=$1
+
+do_trans=true    
+
+
+# All checks passed, ok to prepare directory. but we do not Copy model and other files from
+# the first source.
+
+for src in $@; do
+  if [[ "$(cd 2>/dev/null -P -- "$src" && pwd)" = \
+        "$(cd 2>/dev/null -P -- "$dest" && pwd)" ]]; then
+    echo "$0: error: Source $src is same as target $dest."
+    exit 1
+  fi
+  if $do_trans && [[ ! -f $src/trans.1 ]]; then
+    echo "$0: warning: transform (trans.*) are not present in $src, not" \
+         "combining. please check you files" 
+    exit 1
+  fi
+done
+
+if [ ! -f $dest/ali.1.gz  ] && [ ! -f $dest/lat.1.gz ] ; then 
+    echo "warning: we assume you have combined the ali or lat dirs " \
```

Comment:
make sure it starts with "$0: warning: ..." so its clear which program generated the message